contract/propertyMarketplace.sol

```diff
@@ -84,192 +135,170 @@ contract PropertyMarketplace {
             _houseToken
         );
         propertiesLength++;
-        
-
     }
 
-    function readProperty(uint _index) public view returns (
-        address payable,
-        PropertyLabel memory,
-        PropertyStockData memory,
-        uint, 
-        uint, 
-        Status,
-        HouseToken
-    ) {
-        return (
-            properties[_index].owner,
-            properties[_index].label, 
-            properties[_index].stockData,
-            properties[_index].bedrooms,
-            properties[_index].bathrooms,
-            properties[_index].status,
-            properties[_index].houseToken
-        );
+    /// @dev returns the data of a property
+    function readProperty(uint _index)
+        public
+        view
+        exist(_index)
+        returns (Property memory)
+    {
+        return (properties[_index]);
     }
 
-    function getPropertiesLength()
-    public
-    view
-    returns (uint){
+    function getPropertiesLength() public view returns (uint) {
         return propertiesLength;
     }
 
+    /// @dev returns the number of tokens spender has been approved for
     function getAllowance(address spender, uint _index)
-    public
-    view
-    returns (uint){
-        return properties[_index].houseToken.allowance(properties[_index].owner, spender);
+        public
+        view
+        exist(_index)
+        returns (uint)
+    {
+        return
+            properties[_index].houseToken.allowance(
+                properties[_index].owner,
+                spender
+            );
     }
 
-    function getTokenAddress(uint index)
-    public
-    view
-    returns(address){
-        return address(properties[index].houseToken);
+    /// @dev returns the number of shares of address owner
+    function getBalance(uint _index, address owner)
+        public
+        view
+        exist(_index)
+        returns (uint)
+    {
+        return properties[_index].houseToken.balanceOf(owner);
     }
 
-    function getBalance(uint index,address owner)
-    public
-    view
-    returns(uint){
-        return properties[index].houseToken.balanceOf(owner);
-    }
-    function approveSpender(address spender, uint index) public {
-        properties[index].houseToken.approve(spender, 10);
+    /// @dev approves address spender to use a specified number of tokens
+    function approveSpender(
+        address spender,
+        uint _index,
+        uint amount
+    ) external exist(_index) onlyPropertyOwner(_index) {
```

Comment:
This function can be removed entirely because it's not used.
In the end I decided to let the smart contract of the marketplace to own the House Tokens.

Reply:
That's fine too!